src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpUseSearchValues.Fixer.cs

```diff
@@ -0,0 +1,133 @@
+                declaration.Variables is [var declarator])
+            {
+                var newDeclarator = declarator.ReplaceToken(declarator.Identifier, declarator.Identifier.WithAdditionalAnnotations(RenameAnnotation.Create()));
+                return fieldDeclaration.WithDeclaration(declaration.WithVariables(new SeparatedSyntaxList<VariableDeclaratorSyntax>().Add(newDeclarator)));
+            }
+
+            return node;
+        }
+
+        protected override SyntaxNode GetDeclaratorInitializer(SyntaxNode syntax)
+        {
+            if (syntax is VariableDeclaratorSyntax variableDeclarator)
+            {
+                return variableDeclarator.Initializer!.Value;
+            }
+
+            if (syntax is PropertyDeclarationSyntax propertyDeclaration)
+            {
+                return CSharpUseSearchValuesAnalyzer.TryGetPropertyGetterExpression(propertyDeclaration)!;
+            }
+
+            throw new InvalidOperationException($"Expected 'VariableDeclaratorSyntax' or 'PropertyDeclarationSyntax', got {syntax.GetType().Name}");
+        }
+
+        // new[] { 'a', 'b', 'c' } => "abc"
+        // new[] { (byte)'a', (byte)'b', (byte)'c' } => "abc"u8
+        // "abc".ToCharArray() => "abc"
+        protected override SyntaxNode? TryReplaceArrayCreationWithInlineLiteralExpression(IOperation operation)
+        {
+            if (operation is IConversionOperation conversion)
+            {
+                operation = conversion.Operand;
+            }
+
+            if (operation is IArrayCreationOperation arrayCreation &&
+                arrayCreation.GetElementType() is { } elementType)
+            {
+                bool isByte = elementType.SpecialType == SpecialType.System_Byte;
+
+                if (isByte &&
+                    (operation.SemanticModel?.Compilation is not CSharpCompilation compilation ||
+                    compilation.LanguageVersion < (LanguageVersion)1100)) // LanguageVersion.CSharp11
+                {
+                    // Can't use Utf8StringLiterals
+                    return null;
+                }
+
+                List<char> values = new();
+
+                if (arrayCreation.Syntax is ExpressionSyntax creationSyntax &&
+                    CSharpUseSearchValuesAnalyzer.IsConstantByteOrCharArrayCreationExpression(operation.SemanticModel!, creationSyntax, values, out _) &&
+                    values.Count <= 128 &&                  // Arbitrary limit to avoid emitting huge literals
+                    !ContainsAnyComments(creationSyntax))   // Avoid removing potentially valuable comments
+                {
+                    string valuesString = string.Concat(values);
+                    string stringLiteral = SymbolDisplay.FormatLiteral(valuesString, quote: true);
+
+                    const SyntaxKind Utf8StringLiteralExpression = (SyntaxKind)8756;
+                    const SyntaxKind Utf8StringLiteralToken = (SyntaxKind)8520;
```

Comment:
Is there any side effect of using such enum values @mavasani?

Reply:
This should be fine. These values are not likely to change, and we do not want to end up referencing newer Microsoft.CodeAnalysis versions that do contain these kinds as that will prevent the analyzer NuGet package from working on older VS and compiler versions.